### DISCUSSION TITLE
Multiple registries?

### DISCUSSION THREAD

**Discussion by sporritt** [verdaccio/verdaccio#2309] [Q&A] [answered]
I'm looking to switch over to Verdaccio from Nexus but there's one thing I can't see any evidence of Verdaccio supporting - multiple registries at the root level.  Am I missing something? All of the documentation I've seen show commands like this:

```
npm adduser --registry http://localhost:4873
```

...suggesting there's only one registry.

In Nexus I can create any number of registries and then connect to them:

```
npm adduser --registry http://localhost:8080/foo
npm adduser --registry http://localhost:/bar
```

I need this because I serve different contents for a given package to different people. I have a number of users who all want the package `@mycompany/excellent-thing`, but each of those users should get a different download (not different in terms of functionality, for what it's worth, just containing a different license key). 

Incidentally, I suppose an alternative is some means of identifying, for a given authenticated user, from where in the storage that package should be sourced.  If such a plugin exists or could be built I'm also interested in that.

**Comment by sporritt** [comment]
...further to my last point about a plugin, I guess that would fall into the category of a "storage plugin":

https://verdaccio.org/docs/en/plugin-storage

I'm brand new to this codebase but this method seems like the one that I would want to inject my functionality into:

```
readPackage(fileName: string, callback: ReadPackageCallback): void;
```

and it doesn't take any user information, which would seem to scupper that idea. Any thoughts/suggestions appreciated.

**Comment by juanpicado** [comment, marked as answer]
I don't think verdaccio support such workflow or any plugin that does something like that. 

Regarding the plugin, currently the storage plugin does not have access to the auth, only configuration. Only the middle ware has access two both (auth, storage) instances. Maybe in the future this change but I have no a clear suggestion a this point.

**Comment by sporritt** [comment]
Thanks for getting back to me. Looking at what's available to a middleware plugin, and some of the example use cases, I think a middleware plugin could well be a candidate.
